From e529a463c4badf6549460f13791788b1e3dbdd71 Mon Sep 17 00:00:00 2001
From: Rander Wang <rander.wang@linux.intel.com>
Date: Fri, 6 Jul 2018 17:35:59 +0800
Subject: [PATCH 1326/1677] ASoC: SOF: return status in snd_sof_ipc_reply

The IPC algorithm depends on the status of snd_sof_ipc_reply.
The failed case is processed different with success one

Signed-off-by: Rander Wang <rander.wang@linux.intel.com>
---
 sound/soc/sof/ipc.c      | 5 +++--
 sound/soc/sof/sof-priv.h | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/sound/soc/sof/ipc.c b/sound/soc/sof/ipc.c
index 300f331..6b2b1d1 100644
--- a/sound/soc/sof/ipc.c
+++ b/sound/soc/sof/ipc.c
@@ -247,7 +247,7 @@ void sof_ipc_drop_all(struct snd_sof_ipc *ipc)
 EXPORT_SYMBOL(sof_ipc_drop_all);
 
 /* handle reply message from DSP */
-void snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id)
+int snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id)
 {
 	struct snd_sof_ipc_msg *msg;
 
@@ -255,11 +255,12 @@ void snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id)
 	if (!msg) {
 		dev_err(sdev->dev, "error: can't find message header 0x%x",
 			msg_id);
-		return;
+		return -EINVAL;
 	}
 
 	/* wake up and return the error if we have waiters on this message ? */
 	sof_ipc_tx_msg_reply_complete(sdev->ipc, msg);
+	return 0;
 }
 EXPORT_SYMBOL(snd_sof_ipc_reply);
 
diff --git a/sound/soc/sof/sof-priv.h b/sound/soc/sof/sof-priv.h
index 91115c8..72b8870 100644
--- a/sound/soc/sof/sof-priv.h
+++ b/sound/soc/sof/sof-priv.h
@@ -385,7 +385,7 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 offset);
  */
 struct snd_sof_ipc *snd_sof_ipc_init(struct snd_sof_dev *sdev);
 void snd_sof_ipc_free(struct snd_sof_dev *sdev);
-void snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id);
+int snd_sof_ipc_reply(struct snd_sof_dev *sdev, u32 msg_id);
 void snd_sof_ipc_msgs_rx(struct snd_sof_dev *sdev);
 void snd_sof_ipc_msgs_tx(struct snd_sof_dev *sdev);
 int snd_sof_ipc_stream_pcm_params(struct snd_sof_dev *sdev,
-- 
2.7.4

